### DISCUSSION TITLE
Load test do not complete for some vusers

### DISCUSSION THREAD

**Discussion by yogi-sagar** [artilleryio/artillery#2442] [Artillery]
Hi,
Can someone throw some light if they have seen this issue?
I am trying to run an artillery/playwright script on my application with simple steps of filtering and loading the data on the grid.
I am running the script with 20 vusers.
9 out of 10 times I see that the run get stuck in between without throwing any error.
In one such run - 20 vusers were created & logged in to the system, out of them, 16 vusers made it to half way mark, out of them 12 vusers completed the script successfully.
I do not see any error being generated anywhere.

In rare case, I see all 20 users finishing successfully. So what cause some of the vusers being stuck while few of them get finished successfully? I do not see CPU or Memory build up causing system under stress.

Any idea?

**Comment by hassy** [comment]
hi @yogi-sagar 👋 best thing to do is to enable tracing with one of the supported providers (Datadog, Honeycomb, New Relic, Lightstep, Grafana or another vendor that supports OpenTelemetry). Docs here: https://www.artillery.io/docs/reference/extensions/publish-metrics/opentelemetry

Those traces will give you full visibility into what each VU is doing and any errors.

There's a short demo here: https://www.artillery.io/blog/end-to-end-tracing-for-playwright-tests
